pydantic/functional_validators.py

```diff
@@ -193,7 +193,7 @@ def __get_pydantic_core_schema__(self, source_type: Any, handler: GetCoreSchemaH
             serialization = core_schema.wrap_serializer_function_ser_schema(
                 function=lambda v, h: h(v),
                 schema=schema,
-                return_schema=schema,
+                return_schema=handler.generate_schema(source_type),
```

Comment:
What if we check that `schema` doesn't contain any `serialization` schema in it?

```python
schema = handler(source_type)
if 'serialization' not in schema:
    ...
```

Reply:
Maybe... wouldn't it be more complex though - that doesn't handle nested dicts?

Reply:
I don't think that would work - don't custom serializers just have their own wrap serializer functions?